### ISSUE TITLE
Cannot get subpath configuration to work with nginx reverse proxy

### ISSUE THREAD

**Issue by rexy712** [Kruptein/PlanarAlly#1005]
I have tried following the server installation guides. I've also looked at any issues on here relevant to my setup. I just can't figure out how to get it to run under a subpath. The webpage resolves to a blank white screen.

Everything works fine when running without a subpath though.

I'm running the server with PA_BASEPATH='/subpath/'

My current nginx config:
```
        upstream aiohttp {
                server 127.0.0.1:8000;
        }
        server {
                listen 80;
                listen [::]:80;
                server_name <MY_DOMAIN>;
                return 301 https://$server_name:443$request_uri;
        }
        server {
                listen 443 ssl default_server;
                listen [::]:443 ssl default_server;
                ssl_certificate <MY_CERT>;
                ssl_certificate_key <MY_CERT>;

                server_name <MY_DOMAIN>;



                location /subpath/ {
                        proxy_set_header Host $http_host;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_redirect off;
                        proxy_buffering off;
                        proxy_pass http://aiohttp/subpath;
                }

                location /subpath/socket.io/ {
                        proxy_http_version 1.1;
                        proxy_set_header Upgrade $http_upgrade;
                        proxy_set_header Connection "Upgrade";
                        proxy_set_header Host $host;
                        proxy_pass http://aiohttp/subpath/socket.io/;
                }

                location /subpath/static {
                        alias /var/www/htdocs/PlanarAlly/server/static;
                }
        }
```

**Comment by Kruptein** [comment]
I'll check it out more in-depth tomorrow, as I'm about to go to sleep :smile:,
But can you specify which version of PA you're trying to run? (e.g. a specific branch or release?)

**Comment by rexy712** [comment]
Oh yeah sorry. I'm on tag 2022.2.3.

**Comment by Kruptein** [comment]
Ok cheers!
I also just realised that the docs are not up to date for the subpath stuff.
In 2022.1 I changed some behaviour and apparently I haven't updated the docs properly.

I wrote some info about it in the release notes for that version over here: https://www.planarally.io/blog/release-2022.1/#breaking-subpath-handling

In practice you'll have to build the client with the env variable set as well.

**Comment by rexy712** [comment]
That did the trick! I really should have tested that before posting. I was too busy pulling my hair out thinking I had something wrong in the nginx configs.
Thanks for the super quick responses!

**Comment by rexy712** [comment]
I spoke too soon. The login screen does work now and I can get in to create a new session. But when I join the session, there is no grid, and nothing can be done. I can select tools but they aren't usable on anything. I checked out a managed instance and it loads up a grid as soon as it launches. I tried reverting back to no subpath and it doesn't change anything.

![image](https://user-images.githubusercontent.com/2442544/182046920-4ed42e24-0f4a-4b02-87f0-2655923198c6.png)
